### PR TITLE
Request signing flow + Fiducia adapter (AIS)

### PR DESCRIPTION
**Pull request by n0rdy**

### CODE REVIEW

File: xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/http/RequestSigningInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    private static final List<String> SIGNATURE_HEADERS
+            = Arrays.asList(DIGEST, X_REQUEST_ID, PSU_ID, PSU_CORPORATE_ID, DATE, TPP_REDIRECT_URI);
+
+    private final RequestSigningService requestSigningService = new RequestSigningService();
+
+    @Override
+    public void process(HttpRequest request, HttpContext context) {
+        // Digest header computing and adding MUST BE BEFORE the Signature header, as Digest is used in Signature header computation
+        Digest digest = buildDigest(request);
+        request.addHeader(digest.getHeaderName(), digest.getHeaderValue());
```
Comment (spetrychenko) [inline xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/http/RequestSigningInterceptor.java]: setting header could be encapsulated inside buildDigest method and do the same for below methods

File: xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/signing/service/hashing/ShaHashingService.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package de.adorsys.xs2a.gateway.signing.service.hashing;
+
+import de.adorsys.xs2a.gateway.signing.exception.HttpRequestSigningException;
+import de.adorsys.xs2a.gateway.signing.service.algorithm.HashingAlgorithm;
+
+import java.nio.charset.Charset;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public abstract class ShaHashingService implements HashingService {
```
Comment (spetrychenko) [inline xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/signing/service/hashing/ShaHashingService.java]: Seems it is not `SHA` hashing service, it looks like abstract hashing service

File: adapters/fiducia-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/FiduciaAccountInformationService.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    private static final String BASE_URI = "https://xs2a-test.fiduciagad.de/xs2a/v1";
+
+    @Override
+    protected String getBaseUri() {
+        return BASE_URI;
+    }
+
+    @Override
+    protected Map<String, String> populatePostHeaders(Map<String, String> map) {
+        Map<String, String> headers = super.populateGetHeaders(map);
```
[the remark this answers is not in the document]
Comment (n0rdy) [reply adapters/fiducia-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/FiduciaAccountInformationService.java]: Done, thanks!

File: adapters/fiducia-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/FiduciaAccountInformationService.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        headers.put(DATE_HEADER, DateTimeFormatter.RFC_1123_DATE_TIME.format(ZonedDateTime.now()));
+
+        return headers;
+    }
+
+    @Override
+    protected Map<String, String> populateGetHeaders(Map<String, String> map) {
+        Map<String, String> headers = super.populateGetHeaders(map);
```
[the remark this answers is not in the document]
Comment (n0rdy) [reply adapters/fiducia-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/FiduciaAccountInformationService.java]: Done, thanks!

File: xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/http/RequestSigningInterceptor.java (hunk 1 above)
Comment (n0rdy) [reply xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/http/RequestSigningInterceptor.java]: Done, thanks!

File: xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/signing/service/hashing/ShaHashingService.java (hunk 2 above)
Comment (n0rdy) [reply xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/signing/service/hashing/ShaHashingService.java]: Renamed to `AbstractShaHashingService`, thanks! 
Sha is used in a name as this class should be used for SHA algorithms only. I'm not sure, that the same Java API methods will be used for non-SHA algorithms.

File: adapters/fiducia-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/FiduciaAccountInformationService.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        headers.put(DATE_HEADER, DateTimeFormatter.RFC_1123_DATE_TIME.format(ZonedDateTime.now()));
+
+        return headers;
+    }
+
+    @Override
+    protected Map<String, String> populatePutHeaders(Map<String, String> map) {
+        Map<String, String> headers = super.populateGetHeaders(map);
```
[the remark this answers is not in the document]
Comment (n0rdy) [reply adapters/fiducia-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/FiduciaAccountInformationService.java]: Done, thanks!

**Pull request merged**